src/app/cli/interpreter.cpp

```diff
@@ -1432,20 +1437,47 @@ Interpreter::Value Interpreter::ProcessBr(const Expression &aExpr)
         nlohmann::json                            baJson;
         char                                      mdnsSendBuffer[kMdnsBufferSize];
 
-        if (mContext.mCommandKeys.size() == 2 && mContext.mCommandKeys[0] == "--timeout")
+        for (auto it = mContext.mCommandKeys.begin(); it != mContext.mCommandKeys.end(); ++it)
         {
-            try
+            if (*it == "--timeout")
             {
-                scanTimeout = stol(mContext.mCommandKeys[1]);
-            } catch (...)
+                if (++it != mContext.mCommandKeys.end())
+                {
+                    try
+                    {
+                        scanTimeout = stol(*it);
+                    } catch (...)
+                    {
+                        ExitNow(value = ERROR_INVALID_ARGS("Imparsable timeout value '{}'", *it));
+                    }
```

Comment:
Basically, try to avoid Exception in OT Commissioner code. As an alternative, you can use the c method `strtol()`.
 Additionally, better add a helper method like
```
Error ParseInt(const char* aStr, int &aValue);
```

Reply:
I found we have a dedicated helper method named `emplate <typename T> static Error ParseInteger(T &aInteger, const std::string &aStr)` which could be used here to replace the `Exception` block.